trie.cpp

```diff
@@ -0,0 +1,78 @@
+#include "trie.h"
+#include <algorithm>
+
+void Trie::insertStringInternal(shared_ptr<Node> node, string s, std::size_t strIdx){
```

Comment:
why std::size_t but not std::shared_ptr or std::string? I'd prefer they all use the std::.